### DISCUSSION TITLE
Is it possible to stream audio over local network?

### DISCUSSION THREAD

**Discussion by apoorv569** [MusicPlayerDaemon/MPD#1732] [General]
I am wondering if it is possible to setup a dedicated `VM` and install `MPD` on it give it a `NFS` share of music and connect using a client like `ncmpcpp` from my laptop to listen to music on my laptop.

Is this possible to do so?

**Comment by MaxKellermann** [comment]
That's possible, but not a good idea. MPD is designed to run on the machine that has the sound card and the speakers.
MPD clients only control playback, but they do not actually emit any sound. It's like a remote control - you don't hear sound from your remote control.
Install MPD on your laptop if you want your laptop to play music. Or install any other music player you like, but installing MPD on a VM without any sound card doesn't make sense.

**Reply by apoorv569** [reply]
> That's possible, but not a good idea. MPD is designed to run on the machine that has the sound card and the speakers. MPD clients only control playback, but they do not actually emit any sound. It's like a remote control - you don't hear sound from your remote control. Install MPD on your laptop if you want your laptop to play music. Or install any other music player you like, but installing MPD on a VM without any sound card doesn't make sense.

I see. I only want this for local network I already have another server for external access outside my home (`Airsonic`). This `VM` will help me connect multiple machines from a single `MPD` server local only, otherwise I have to allow multiple machines access to the `NFS` share and configure `MPD` on multiple machines as well.

**Comment by martadinata666** [comment]
I also the one that use MPD as LAN playback, set MPD output to http, and stream the http using mpv or vlc `on phone, or other pc`.

**Reply by apoorv569** [reply]
> I also the one that use MPD as LAN playback, set MPD output to http, and stream the http using mpv or vlc `on phone, or other pc`.

I want to use `ncmpcpp`. I have tried `http` output with `mpv` it does work but then I have to open multiple programs to listen to audio that doesn't makes sense to me.

**Reply by martadinata666** [reply]
there is gmpc with shout plugins that can help start the stream on bg, if you want all in one thing but a gui indeed. ncmpcpp is only controlling playback, not really sure if there is hacky way to make it stream. ☕
